sql/src/main/kotlin/io/github/codebandits/beak/entityOperations.kt

```diff
@@ -1,21 +1,32 @@
 package io.github.codebandits.beak
 
 import arrow.core.Either
+import arrow.core.flatMap
 import arrow.data.Try
 import org.jetbrains.exposed.dao.Entity
 import org.jetbrains.exposed.dao.EntityClass
 import org.jetbrains.exposed.sql.transactions.TransactionManager
 
-fun <ID : Comparable<ID>, T : Entity<ID>> EntityClass<ID, T>.allOrError(): Either<DataAccessError.SystemError, List<T>> =
-    Try { all().toList() }.handleDataAccessErrors()
+fun <ID : Comparable<ID>, T : Entity<ID>> EntityClass<ID, T>.allOrError(): Either<DataAccessError, List<T>> =
+    Try { all().toList() }.mapFailureToDataAccessError()
 
 fun <ID : Comparable<ID>, T : Entity<ID>> EntityClass<ID, T>.newOrError(init: T.() -> Unit): Either<DataAccessError, T> =
-    Try { new(init).also { TransactionManager.current().commit() } }.handleDataAccessErrors()
+    Try { new(init).also { TransactionManager.current().commit() } }.mapFailureToDataAccessError()
 
-private fun <T> Try<T>.handleDataAccessErrors(): Either<DataAccessError.SystemError, T> = toEither().mapLeft {
+fun <ID : Comparable<ID>, T : Entity<ID>> EntityClass<ID, T>.findByIdOrError(id: ID): Either<DataAccessError, T> =
+    Try { findById(id) }
+        .mapFailureToDataAccessError()
+        .flatMap {
```

Comment:
I refactored the handling of null to just use flatMap because it felt simpler to me. Do you like it @rodolfo2488?